### DISCUSSION TITLE
Set varchar limit in UI, SQL, or let Supabase optimize?

### DISCUSSION THREAD

**Discussion by jmlivingston** [supabase/supabase#5885] [Questions] [answered]
Is there a way to set the limit with **varchar** in the UI or is this something I need to do through SQL? 

Or, is this something I shouldn't worry about since Supabase optimizes this on the fly?

**Comment by GaryAustin1** [comment, marked as answer]
You normally would not worry about it (use it) at all using Postgres.
https://www.postgresql.org/docs/14/datatype-character.html
https://wiki.postgresql.org/wiki/Don%27t_Do_This#Don.27t_use_varchar.28n.29_by_default

If you are trying to force a limit on length for some reason you probably are better off adding a constraint to the column.

**Reply by jmlivingston** [reply]
@GaryAustin1 - Thanks for sharing the links. Most of my SQL experience has been with the MS variety where it is encouraged, so this was very helpful.
